src/google/protobuf/compiler/java/java_message.cc

```diff
@@ -124,7 +124,7 @@ void ImmutableMessageGenerator::GenerateStaticVariables(io::Printer* printer) {
 
     // The descriptor for this type.
     printer->Print(vars,
-      "$private$static final com.google.protobuf.Descriptors.Descriptor\n"
+      "$private$static com.google.protobuf.Descriptors.Descriptor\n"
```

Comment:
Never mind. I just want to understand why it's removed and the reason you gave is good enough to me. Could you fix my other comment? I'll merge the change after you update that.